### PR TITLE
feat: add build-isolation options

### PR DESCRIPTION
**Pull request by tdejager**
Adds the ability to do build-isolation. 

Would be good to get in: https://github.com/astral-sh/uv/pull/6517/, so it's more useful in issue like: https://github.com/prefix-dev/pixi/issues/1880 and https://github.com/prefix-dev/pixi/issues/1834

**edit**:
The 6517 PR mentioned above has been integrated, started UV upgrade here: https://github.com/prefix-dev/pixi/pull/1936 which we need to have the added test to succeed.

**edit**
All prequisites are in.

TODO:
- [x] Documenation 
- [x] Json Schema
- [x] Rust test
- [x] (nice to have) One example that uses it

### CODE REVIEW

**Comment by tdejager** [comment]
Would be good if could start reviewing this @baszalmstra @ruben-arts

File: schema/model.py [hunk 1]
```diff
@@ -497,6 +497,12 @@ class PyPIOptions(StrictBaseModel):
         description="Paths to directory containing",
         examples=[["https://pypi.org/simple"]],
     )
+    no_build_isolation: list[PyPIPackageName] = Field(
```
Comment (ruben-arts) [inline schema/model.py:500]: Can you add it to the tests for the schema. (`examples/valid/full.toml`)

**Comment by tdejager** [comment]
@ruben-arts also added an example that builds/installs detectron2 from source.

File: schema/model.py (hunk 1 above)
Comment (tdejager) [reply schema/model.py:500]: Done.

**Review by ruben-arts — APPROVED** [review]

**Comment by ruben-arts** [comment]
Thanksss

**Pull request merged**